pkg/codegen/testing/test/testdata/third-party-package-pp/go/third-party-package.go

```diff
@@ -20,6 +19,12 @@ func main() {
 		if err != nil {
 			return err
 		}
+		_, err := other.NewProvider(ctx, "Provider", &other.ProviderArgs{
+			Username: pulumi.String("foo"),
+		})
```

Comment:
Interesting. This has revealed another bug.

We're not importing the name `other` anywhere here. I'd expect one of two things:

1. This is `thirdparty.NewProvider`, and we keep the import `git.example.org/thirdparty`
2. This is `other.NewProvider` and we alias the import as `other`.

Reply:
[The pkg name](https://github.com/pulumi/pulumi/blob/0287fb0a5eceb405adf2f7329f31a1065c239c9a/pkg/codegen/testing/test/testdata/other.json) is `other` and the import is `thirdparty`, so #1 is expected. Would we expect to translate `other:index:Thing` -> `import git.example.org/thirdparty/index` and have `index.NewThing`?

Reply:
[The pkg name](https://github.com/pulumi/pulumi/blob/8e9f405fc777932d6737e06efcdd9fa100163b85/pkg/codegen/testing/test/testdata/other.json) is `other` but the import path is "git.example.org/thirdparty", so I think we need #2

Reply:
I don't think this is a bug – in [other.json](https://github.com/pulumi/pulumi/blob/8e9f405fc777932d6737e06efcdd9fa100163b85/pkg/codegen/testing/test/testdata/other.json), we didn't have any alias, and we set the mod of `other:index:Thing` to be `other` correctly, hence `other.NewThing`, while the basepath is `git.example.org/thirdparty`. I'm not sure what the intention of setting the module to be `index` was to hit this special case, but in the meantime I added the `other` alias to the resource schema cc @iwahbe

Reply:
I don't think we should care about the package name. We should see #1 (`third party.NewProvider`).

Regardless, this is unrelated to aliases in the schema, those should never be required for correct imports (any good schema should generate valid code). If we want to care about the package name, the aliasing would be 
```go
package sdk
import (
    other "git.example.org/thirdparty"
)

other.NewProvider(...)
```

Reply:
Our [expected examples](https://github.com/pulumi/pulumi/tree/master/pkg/codegen/testing/test/testdata/third-party-package-pp) for all other languages import using the alias `other`. I'd expect consistent behavior, so if we expect #1, is there an issue with those languages' codegen?

Reply:
For posterity here, the channel consensus and some intentional design choices (such as with aws-native) lean toward using the Go package name (#1) without an alias, and not the Pulumi package name.